modules/meshrenderinggl/glsl/oit/sort.glsl

```diff
@@ -105,27 +105,33 @@ void fillFragmentArraySortedUntilDepth(uint pixelIdx, float maxDepth, out int nu
     // Load fragments into a local memory array for sorting
     numFrag = 0;
     float depth = 0.0;
+    uint lastPtr = 0;
 
     while (numFrag < ABUFFER_SIZE) {
-        vec4 val = selectionSortNext(pixelIdx, depth);
+        vec4 val = selectionSortNext(pixelIdx, depth, lastPtr);
         depth = val.y;
         if (depth < 0.0 || depth > maxDepth) return;
         fragmentList[numFrag] = val;
         numFrag++;
     }
 }
 
-vec4 selectionSortNext(uint headPtr, float lastDepth) {
+vec4 selectionSortNext(uint headPtr, float lastDepth, inout uint lastPtr) {
     vec4 closestVal = vec4(-1.0);
     float minDepth = 1.0;
+    bool passedLastPtr = false;
+    uint minPtr = 0;
     while (headPtr != 0) {
         vec4 val = readPixelStorage(headPtr - 1);
-        if (val.y > lastDepth && val.y < minDepth) {
+        if (headPtr == lastPtr) passedLastPtr = true;
```

Comment:
No need for the extra bool. Just skip if they are equal

Reply:
See other comment.

Reply:
Should not this suffice:
if (val.y < minDepth && (val.y > lastDepth || (headPtr > lastPtr && val.y == lastDepth) ))

Reply:
We can't be sure the pointer is in the same order as the linked list. I guess it would mostly be, but it's not a given.